packages/components/calendar/src/index.vue

```diff
@@ -237,6 +258,10 @@ export default defineComponent({
         day = prevMonthDayjs.value
       } else if (type === 'next-month') {
         day = nextMonthDayjs.value
+      } else if (type === 'prev-year') {
+        day = prevYearDayjs.value
+      } else if (type === 'next-year') {
+        day = nextYearDayjs.value
       } else {
```

Comment:
Also I think you are doing **2** things inside **1** PR which is not recommended. Please revert all changes for adding `prev-year``next-year` and file another PR for this. Thank you!